src/pattern.py

```diff
@@ -1,2 +1,14 @@
 
 # Print the pattern
+
+pattern = "*"
+
+for i in range(9):
+    if i==0:
+        print(pattern)
+    elif i<5:
+        pattern = pattern + " *"
+        print(pattern)
+    else:
+        pattern = pattern[:-2]
+        print(pattern)
```

Comment:
This is a fine solution. For reasons that we haven't gotten to, though, appending to strings and slicing off them is not as efficient as you would want. But that is no criticism of you; we just haven't made it there in the class.